src/xtgeo/surface/_regsurf_gridding.py

```diff
@@ -322,21 +322,26 @@ def surf_fill(self, fill_value=None):
         logger.info("Do fill... DONE")
 
 
-def smooth_median(self, iterations=1, width=1):
-    """Smooth a surface using a median filter.
+def _smooth(
+    self,
+    window_function: Callable[[np.ndarray], np.ndarray],
+    iterations: int = 1,
+    width: int = 1,
```

Comment:
I don't like the default value of one, as it would not change the input. I think 3 or 5 are "sane" default values.